hibernate-rx-core/src/main/java/org/hibernate/rx/engine/impl/Cascade.java

```diff
@@ -395,17 +395,18 @@ private void cascadeCollection(
 			cascadePoint = CascadePoint.AFTER_INSERT_BEFORE_DELETE_VIA_COLLECTION;
 		}
 
+		// TODO: Collection of elements is currently not supported
 		//cascade to current collection elements
-		if ( elemType.isEntityType() || elemType.isAnyType() || elemType.isComponentType() ) {
-			cascadeCollectionElements(
-				componentPathStackDepth,
-				child,
-				type,
-				style,
-				elemType,
-				persister.isCascadeDeleteEnabled()
-			);
-		}
+//		if ( elemType.isEntityType() || elemType.isAnyType() || elemType.isComponentType() ) {
+//			cascadeCollectionElements(
+//				componentPathStackDepth,
+//				child,
+//				type,
+//				style,
+//				elemType,
+//				persister.isCascadeDeleteEnabled()
+//			);
+//		}
```

Comment:
Why is this code disabled? Cascade persist for `@OneToMany` is pretty popular.

Reply:
OK, I see, so here's the thing: that code isn't only for `@ElementCollection`, it's actually for all kinds of collections including `@OneToMany`s.

Unfortunately, `CascadeTest` was missing an assertion, which means the test didn't fail when it really should have.

I've pushed the missing assertion [here](54ccaa07397d59d94a847b90f8d74a13fd2b20da).

Reply:
Now, if I uncomment that code, it *almost* works, except that `CascadeTest` fails when it tries to cascade delete an unfetched many-to-one association. That looks reasonably fixable and I'm looking into it. For now you can make the test pass by adding:

    .thenCompose(v-> s3.fetch(node.parent.elements))

 at line 98 of `CascadeTest`.

Reply:
Thanks, I will look into it

Reply:
@DavideD note that the problem is avoided for to-one associations via the call to `fetch()` at line 52 of `CascadingActions`. We just need something like that for collections as well. The only problem is I'm not quite sure _where_ to add it.

Reply:
> That looks reasonably fixable and I'm looking into it.

If you are working on cascade delete of an unfetched many-to-one association. could you create a new issue, please?

Reply:
Well I think we need it in this changeset, because I added a (currently-passing) test to master that will probably fail with this changeset.

Reply:
@DavideD try [this commit](https://github.com/gavinking/hibernate-rx/commit/6aa11cae058a9082b371de588dd048e4ae304cee), it might be all you need.